File: python_aoc/day_01a.py
left = []
right = []
with open("day_01.txt") as input:
    for line in input:
        left.append(int(line[0:5]))
        right.append(int(line[8:13]))

right.sort()
left.sort()

total_distance = 0
for l, r in zip(left, right):
    total_distance += abs(l - r)

print(total_distance)

File: haskell_aoc/day_01a.hs
import Data.List

main = do
  contents <- readFile "day_01.txt"
  print $ solution contents

solution :: String -> Int
solution input =
  let xs = lines input
      lefts = sort $ map (read . head . words) xs
      rights = sort $ map (read . last . words) xs
   in sum $ zipWith (\l r -> abs (l - r)) lefts rights